### ISSUE TITLE
bug: on_result_diagnostics_quickfix does not populate quickfix list

### ISSUE THREAD

**Issue by EnDeRGeaT** [stevearc/overseer.nvim#392]
### Neovim version (nvim -v)

0.10.3

### Operating system/version

Arch Linux

### Describe the bug

I have added `on_result_diagnostics_quickfix` in my default components. However when I do `:OverseerRunCmd g++ a.cpp` with a file that has errors, my quickfix list is not populated. Yet, when I open output in quickfix list from toggle menu (or i have `on_output_quickfix` component), it matches everything.

### What is the severity of this bug?

breaking (some functionality is broken)

### Steps To Reproduce

1. I setup overseer using `lazy.nvim` with the following code:
```lua
return {
  'stevearc/overseer.nvim',
  opts = {},
  config = function ()
      require('overseer').setup({
          component_aliases = {
              default = {
                  { "display_duration", detail_level = 2 },
                  "on_output_summarize",
                  "on_exit_set_status",
                  "on_complete_notify",
                  "on_result_diagnostics",
                  { "on_result_diagnostics_quickfix", open = true},
              },
          },
      })
  end
}
```
2. Create a.cpp:
```cpp
int main() {
    reutrn 0; // a.cpp:2:5: error: ‘reutrn’ was not declared in this scope
}
```
3. Do `:OverseerRunCmd g++ a.cpp`

### Expected Behavior

quickfix list gets populated

### Additional context

Now thinking about it while writing this issue, is this intended behavior? That's a bit of a shame, I wanted to make a simple task which builds and runs a user provided shell command, and if I get errors (compilation errors), I can go to them via quickfix list.

**Comment by stevearc** [comment]
`on_result_diagnostics_quickfix` is for when you have another component that will parse the output into some result diagnostics (e.g. `on_output_parse`). I believe you're just looking for `on_output_quickfix`. Is there a reason that component doesn't work for you?

**Comment by EnDeRGeaT** [comment]
`on_output_quickfix` doesn't work for me because I have a `on_open_output` component too. I wanted to make it so that if I have errors, my quickfix list opens, otherwise a terminal opens. Currently, I have just made a keymap, which loads output in quickfix list. It's a bit manual but gets the job done. I assume that it is kinda difficult to make `on_output_quickfix` and `on_open_output` work together well, so you can probably close this issue

**Comment by stevearc** [comment]
I think we can get you most of the way there. Try modifying your `on_output_quickfix` component to set `open_on_exit = "failure"`. You may also want `set_diagnostics = true` and/or `items_only = true`. Then for `open_output`, set `on_complete = "success"`. The combination of these two will open the quickfix if there are errors (technically, if the command has a nonzero exit code), and open the terminal output if it succeeds.

**Comment by EnDeRGeaT** [comment]
Oh, that is very nice. Thank you. By the way, is there any way to change the size of `open_output` terminal? I saw that there is a `terminal` strategy with `size`, but there is nothing like that for `open_output`.

**Comment by stevearc** [comment]
No, `open_output` doesn't support that at the moment. I'd be willing to review a PR that added that capability, though
